### PR TITLE
Reading events from splunk search as sent from ufws

### PR DESCRIPTION
**Pull request by ghoto**
Downloading events from Splunk's server API.
I wanted to give Splunk's SDK a shot but it's awfully slow. The `ResultReader` will burn your computer 🔥.
Instead I will just connect to the API with a plain http connection (curl gets the results in a blink of an eye).
Stay tuned for the changes.

### CODE REVIEW

File: modules/TerraformController.py [hunk 1]
```diff
@@ -196,11 +197,11 @@ def dump_attack_data(self, dump_name):
...
-        servers = []
-        if self.config['windows_domain_controller'] == '1':
-            servers.append('windows_domain_controller')
-        if self.config['windows_server'] == '1':
-            servers.append('windows_server')
+        servers = ['splunk_server']
+        # if self.config['windows_domain_controller'] == '1':
+        #     servers.append('windows_domain_controller')
+        # if self.config['windows_server'] == '1':
+        #     servers.append('windows_server')
```
Comment (ghoto) [inline modules/TerraformController.py:204]: This will go uncommented. Just have it here because I was testing.

- [ ] Remove comments

**Comment by ghoto** [comment]
This version DOES NOT use Splunk SDK since it's extremely slow. It just use a direct connection to Splunk server API to extract the search.

**Comment by P4T12ICK** [comment]
Can we add two parameters which control the dump command?
dump_search: 'index=win sourcetype=*sysmon*'
dump_time: '-2h@h'
The first one makes it flexible what you want to dump by defining it as search. The second parameter specifies how long you want to search into the past.

**Comment by ghoto** [comment]
> Can we add two parameters which control the dump command?

Do you want them to be parameters you pass in the command line?

> dump_search: 'index=win sourcetype=_sysmon_'
> dump_time: '-2h@h'
> The first one makes it flexible what you want to dump by defining it as search. The second parameter specifies how long you want to search into the past.

For the dump_time I can add that parameter to the array that defines the dump. I thought it wasn't necessary as the other dumps dump the whole logs, because for making these datasets we were tearing down the range for each of them.

**Comment by P4T12ICK** [comment]
I think this both parameters should be defined in the ttack_range.conf file in the attacK_data_dump section. Let's add the dump time parameter. We will probably need it in the future.

**Comment by P4T12ICK** [comment]
Thank you

**Pull request closed without merging**